tools/WorkerHarness/src/WorkerHarness.Core/WorkerProcess/WorkerProcessBuilder.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using WorkerHarness.Core.Worker;
+
+namespace WorkerHarness.Core.WorkerProcess
+{
+    public class WorkerProcessBuilder : IWorkerProcessBuilder
+    {
+        /// <summary>
+        /// Build an instance of a worker process
+        /// </summary>
+        /// <param name="workerDescription">a WorkerDescription object that contains path info about a language worker</param>
+        /// <param name="workerId">a language worker Id</param>
+        /// <param name="requestId">a request Id</param>
+        /// <returns></returns>
+        /// <exception cref="MissingMemberException"></exception>
+        public Process Build(WorkerDescription workerDescription, string workerId, string requestId)
+        {
+            string workerFile = workerDescription.DefaultWorkerPath ?? throw new MissingMemberException("The default worker path is null");
+            string arguments = $"{workerFile} --host {WorkerProcessConstants.DefaultHostUri} --port {WorkerProcessConstants.DefaultPort} --workerId {workerId} --requestId {requestId} --grpcMaxMessageLength {WorkerProcessConstants.GrpcMaxMessageLength}";
```

Comment:
do we know how the language worker process uses the `--workerId` argument value? I do not remember seeing that being used in dotnet worker.

Reply:
I think the host assigns a 'workerId' to a worker process by including it in the arguments. I know that when the worker sends a StartStream message to the host, it set the WorkerId field in the StartStream message. I don't know if the worker set the WorkerId field to be a new Guid or use the 'workerId' in the process argument.

[GrpcWorkerStartupOptions](https://github.com/Azure/azure-functions-dotnet-worker/blob/main/src/DotNetWorker.Grpc/GrpcStartupOptions.cs)

[How GrpcWorkerStartupOptions is populated?](https://github.com/Azure/azure-functions-dotnet-worker/blob/1c482c4831df1be88b470f776194e34618035d58/src/DotNetWorker.Grpc/GrpcServiceCollectionExtensions.cs#L60)